swift/obj/diskfile.py

```diff
@@ -1322,6 +1322,9 @@ def pickle_async_update(self, device, account, container, obj, data,
             os.path.join(async_dir, ohash[-3:], ohash + '-' +
                          Timestamp(timestamp).internal),
             tmp_dir)
+
+        self.logger.increment('async_pendings.' + account + '.' + container)
```

Comment:
Can you move that to the else branch for checking the `.expiring_objects`. this new metric is only needed in that case.